### PR TITLE
Add zero() to fungible_asset module

### PR DESCRIPTION
**Pull request by movekevin**
### Description
It's very common in Defi to have the follow pattern:
```
let zero = fungible_asset::zero(metadata);
while (some conditions) {
  // Loop through a series of positions and calculate interest
  let interest = fungible_asset::extract(&mut position, interest_amount);

  // Take interest
  fungible_asset::merge(&mut zero, interest);
}
```

It's very hard to do this without zero(). We can add the fungible assets to a vector and then merge them at the end but this is just unnecessarily extra work that costs extra gas (once to insert into a vector and another one when passing through to merge).

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/fungible_asset.move [hunk 1]
```diff
@@ -466,6 +466,16 @@ module aptos_framework::fungible_asset {
         deposit_with_ref(transfer_ref, to, fa);
     }
 
+    /// Create a fungible asset with zero amount.
+    /// This can be useful when starting a series of computations where the initial value is 0.
+    public fun zero<T: key>(metadata: Object<T>): FungibleAsset {
+        let metadata = object::convert<T, Metadata>(metadata);
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/fungible_asset.move:472]:
```suggestion
        let metadata = object::convert(metadata);
```
that shouldn't be necessary.

File: aptos-move/framework/aptos-framework/sources/fungible_asset.move [hunk 2]
```diff
@@ -466,6 +466,16 @@ module aptos_framework::fungible_asset {
...
+        FungibleAsset {
+            metadata,
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/fungible_asset.move:474]: then you could just call object::convert here...

**Review by davidiw — APPROVED** [review]

**Review by lightmark — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/fungible_asset.move (hunk 1 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/fungible_asset.move]: Updated

File: aptos-move/framework/aptos-framework/sources/fungible_asset.move (hunk 2 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/fungible_asset.move]: Updated

**Pull request merged**